### PR TITLE
Implementation of SelectiveSearchSegmentation

### PR DESCRIPTION
**Pull request by the-glu**
This is a proposition of an implementation of a segmentation algorithm, described in [Selective search for object recognition](http://koen.me/research/pub/uijlings-ijcv2013-draft.pdf).

It use #361 as a base algorithm, I put it in the same module, I don't know if it's ok but I can move it :)

### CODE REVIEW

File: modules/ximgproc/include/opencv2/ximgproc/segmentation.hpp [hunk 1]
```diff
@@ -117,6 +296,67 @@ namespace cv {
 
             };
 
+            // SelectiveSearch helpers
```
Comment (alalek) [inline modules/ximgproc/include/opencv2/ximgproc/segmentation.hpp]: It is better to move implementation-specific declarations from public header file. You can place them in .cpp file.

**Comment by the-glu** [comment]
I moved helpers from the headers to the cpp, updated the readme and (should have) fixed the warnings in non-linux builds :)

**Comment by alalek** [comment]
Thanks!

You need to change location of the sample (compilation of this sample is not tested now):
```
git mv modules/ximgproc{/src,}/selectivesearchsegmentation.cpp
```

Could you please squash commits into one after all fixes (to remove unnecessary intermediate changes from the git history).

**Comment by the-glu** [comment]
The command doesn't seems to move the location of the sample. Do you want me to move them from `samples/cpp/` to `samples` ?

**Comment by alalek** [comment]
@the-glu Right

**Comment by alalek** [comment]
add "static" to `hsv_to_rgb()` and `color_mapping()`

**Comment by the-glu** [comment]
Yep I'm fixings warnings and I will squash everything when everything is green :P

**Comment by the-glu** [comment]
Okay, everything should be ok now :)

**Comment by alalek** [comment]
:+1:

**Pull request merged**